sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonDataSourceSuite.scala

```diff
@@ -231,8 +212,9 @@ class PythonDataSourceSuite extends QueryTest with SharedSparkSession {
     val schema = StructType.fromDDL("id INT, partition INT")
     val dataSource = createUserDefinedPythonDataSource(
       name = dataSourceName, pythonScript = dataSourceScript)
+    spark.dataSource.registerPython(dataSourceName, dataSource)
```

Comment:
why do we need the extra registration?

Reply:
Previously `UserDefinedPythonDataSource` was able to create a `DataFrame` directly (from `LogicalRelation`) via `UserDefinedPythonDataSource.apply`.

Now it is not possible anymore because we're using DSv2. So, now here we register and load via using `DataFrameReader` to create a `DataFrame` to test.